tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -31,6 +31,8 @@ using RuntimeArgs = std::vector<std::variant<Buffer*, uint32_t>>;
 
 }  // namespace v0
 
+struct Allocator;
+
```

Comment:
There were some recent changes to buffer/allocation recently and I don't think this is required anymore, there were just some things that were missed in the cleanup.

We can remove the `AllocBufferMetadata` struct definition, and the only remaining user of it, `std::optional<AllocBufferMetadata> alloc_md;` from `CommandInterface`.

Reply:
Thanks for flagging this @tt-aho.

I incorporated that cleanup into this PR, and I saw build went through fine.